websites/W/Warriorcats/presence.ts

```diff
@@ -0,0 +1,270 @@
+
+	if (document.location.pathname.startsWith("/staffel-")) {
+		presenceData.details = `Viewing Staffel ${
+			document.location.pathname.split("/")[1].split("-")[1]
+		}`;
+
+		presenceData.largeImageKey =
+			"https://www.warriorcats.de/wp-content/uploads/9783407759085-web.jpg";
+		presenceData.buttons = [
+			{ label: "Current Page", url: document.location.href },
+			{ label: "Warrior Cats Germany", url: "https://www.warriorcats.de/" },
+		];
+	}
+
+	// https://www.warriorcats.de/beute/staffel-viii-band-2-wolken/
+	// extract Staffel number and book number and book name from URL
+	// like this Staffel 8 Band 2 Wolken
+	// and then set the presence state to Currently Staffel 8 Band 2 Wolken
+
+	async function getStaffelName(postNumber: string) {
+		const staffelName = document.querySelector(
+				`#post-${postNumber} > div > div.entry-content > div > div.wc-biblio > div.wc-staffel`
+			).textContent,
+			// Remove "Warrior Cats |"
+
+			// remove Staffel
+
+			// get staffel number from url e.g
+
+			// for ressult3 match replacethis with result2 and replacce it with nothing
+			result3 = staffelName
+				.replace(/^Warrior Cats \| /, "")
+				.replace(/Staffel /, "")
+				.replace(
+					document.location.pathname.split("/")[2].split("-")[1].toUpperCase(),
+					""
+				);
+		// now remove everything that is not a letter or a space and trim the start and end
+		return result3.replace(/[^a-zA-Z ]/g, "").trim();
+	}
+
+	// async function getStaffelNumber(postNumber: string) {
+	//   let staffelNumber = document.querySelector(`#post-${postNumber} div > div.entry-content > div > div.wc-biblio > div.wc-staffel`).textContent;
+	// replace everything after :
+	//    staffelNumber = staffelNumber.replace(/:.*/, "");
+	//     return staffelNumber;
+	//  }
+
+	async function getBuchName(postNumber: string) {
+		let buchName = document.querySelector(
+			`#post-${postNumber} > div > div.entry-content > div > div.wc-biblio > div.wc-band`
+		).textContent;
+		// remove everything before :
+		buchName = buchName.replace(/.*:/, "");
+		return buchName;
+	}
+
+	async function getBuchImage(postNumber: string) {
+		const buchImage = document
+			.querySelector(
+				`#post-${postNumber} > div > div.entry-content > div > div.wc-article-thumbnail > a > img`
+			)
+			.getAttribute("src");
+		return buchImage;
+	}
+
+	if (
+		document.location.pathname.startsWith("/beute/staffel-") ||
+		document.location.pathname.startsWith("/beute/")
+	) {
+		const postNumber = await getPostNumber(),
+			staffelName = await getStaffelName(postNumber),
+			buchName = await getBuchName(postNumber),
+			buchImage = await getBuchImage(postNumber);
+
+		presenceData.state = `Viewing ${buchName}`;
+		presenceData.details = `${staffelName}`;
+		presenceData.largeImageKey = buchImage;
+		presenceData.buttons = [
+			{ label: "Current Page", url: document.location.href },
+			{ label: "Warrior Cats Germany", url: "https://www.warriorcats.de/" },
+		];
+	}
+	presence.clearActivity();
```

Comment:
Since the activity is set immediately afterwards, this is not needed, and could cause problems.
```suggestion
```